### PR TITLE
Feature/ Challenge based on rounds

### PR DESCRIPTION
**Pull request by Jayashsatolia403**

### CODE REVIEW

**Review by dabasov — APPROVED** [review]
lgtm

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -79,7 +79,7 @@ jobs:
           github-token: ${{ github.token }}
 
       - name: "Set PR status as pending"
-        uses: 0chain/actions/set-pr-status@master
+        uses: 0chain/actions/set-pr-status@fix/challenge
```
Comment (Kishan-Dhakan) [inline .github/workflows/ci.yml:82]: revert

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -109,9 +109,9 @@ jobs:
 
       - name: "Deploy 0Chain"
         if: github.event_name == 'push' || github.event.inputs.existing_network == ''
-        uses: 0chain/actions/deploy-0chain@master
+        uses: 0chain/actions/deploy-0chain@fix/challenge
```
Comment (Kishan-Dhakan) [inline .github/workflows/ci.yml:112]: revert

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -109,9 +109,9 @@ jobs:
...
         with:
-          repo_snapshots_branch: "${{ env.REPO_SNAPSHOTS_BRANCH }}"
+          repo_snapshots_branch: fix/challenge
```
Comment (Kishan-Dhakan) [inline .github/workflows/ci.yml:114]: revert

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -123,7 +123,7 @@ jobs:
           svc_account_secret: ${{ secrets.SVC_ACCOUNT_SECRET }}
 
       - name: "Run System tests"
-        uses: 0chain/actions/run-system-tests@master
+        uses: 0chain/actions/run-system-tests@fix/challenge
```
Comment (Kishan-Dhakan) [inline .github/workflows/ci.yml:126]: revert

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -143,7 +143,7 @@ jobs:
 
       - name: "Set PR status as ${{ job.status }}"
         if: ${{ (success() || failure()) && steps.findPr.outputs.number && github.event.inputs.test_file_filter == '' }}
-        uses: 0chain/actions/set-pr-status@master
+        uses: 0chain/actions/set-pr-status@fix/challenge
```
Comment (Kishan-Dhakan) [inline .github/workflows/ci.yml:146]: revert

**Review by Kishan-Dhakan — COMMENTED** [review]
General reminder to revert ci pointers and variables, lgtm otherwise, happy to merge with other related PRs

**Pull request merged**